src/main/java/com/yahoo/bullet/rest/pubsub/PubSubService.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright 2017, Yahoo Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.rest.pubsub;
+
+import com.yahoo.bullet.pubsub.PubSubMessage;
+import org.springframework.stereotype.Service;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+@Service
+public class PubSubService {
+    private ConcurrentLinkedQueue<PubSubMessage> queries = new ConcurrentLinkedQueue<>();
+    private ConcurrentLinkedQueue<PubSubMessage> results = new ConcurrentLinkedQueue<>();
+
+    /**
+     * Get the next query from the query queue. Queries are removed after being read once.
+     *
+     * @return The next query.
+     */
+    public String getQuery() {
+        return queries.isEmpty() ? null : queries.poll().asJSON();
+    }
+
+    /**
+     * Get the next result from the result queue. Results are removed after being read once.
+     *
+     * @return the next result.
+     */
+    public String getResult() {
+        return results.isEmpty() ? null : results.poll().asJSON();
+    }
+
+    /**
+     * Add a result to the result queue.
+     *
+     * @param result The result to add to the result queue.
+     */
+    public void postResult(String result) {
+        results.add(PubSubMessage.fromJSON(result));
```

Comment:
Can we have a private method addToQueue(ConcurrentLInkedQueue, String) that does this and use it for both methods?